api/src/backend/api/models.py

```diff
@@ -1222,3 +1222,64 @@ class Meta:
                 statements=["SELECT", "INSERT", "UPDATE", "DELETE"],
             ),
         ]
+
+
+class LighthouseConfig(RowLevelSecurityProtectedModel):
+    """
+    Stores configuration and API keys for LLM services.
+    """
+
+    id = models.UUIDField(primary_key=True, default=uuid4, editable=False)
+    inserted_at = models.DateTimeField(auto_now_add=True, editable=False)
+    updated_at = models.DateTimeField(auto_now=True, editable=False)
+
+    name = models.CharField(max_length=100, validators=[MinLengthValidator(3)])
+    api_key = models.BinaryField(blank=False, null=False)
+    model = models.CharField(max_length=50, default="gpt-4o")
+    temperature = models.FloatField(default=0.7)
+    max_tokens = models.IntegerField(default=4000)
+    business_context = models.TextField(
+        blank=True,
+        null=True,
+        help_text="Additional business context for this AI model configuration",
+    )
+    is_active = models.BooleanField(default=True)
+
+    def __str__(self):
+        return self.name
+
+    @property
+    def api_key_decoded(self):
+        """Return the decrypted API key."""
+        try:
+            return fernet.decrypt(self.api_key).decode()
+        except Exception:
+            return None
```

Comment:
I think that since the user can add this value it is important to verify and track what is happening, I propose a more secure version of this function with more information to detect anomalous cases:
```suggestion
        try:
            decrypted_key = fernet.decrypt(self.api_key)
            return decrypted_key.decode()
        except InvalidToken:
            logger.warning("Failed to decrypt API key: invalid token.")
            return None
        except Exception as e:
            logger.error(f"Unexpected error while decrypting API key: {e}")
            return None
```